modules/dataControllerModule/index.js

```diff
@@ -0,0 +1,194 @@
+      let segments = userData[i].segment;
+      let segmentPrefix = '';
+      if (userData[i].name) {
+        segmentPrefix = userData[i].name + ':';
+      }
+
+      if (userData[i].ext && userData[i].ext.segtax) {
+        segmentPrefix += userData[i].ext.segtax + ':';
+      }
+      for (let j = 0; j < segments.length; j++) {
+        segmentSet.add(segmentPrefix + segments[j].id);
+      }
+    }
+  }
+
+  return segmentSet;
+}
+
+function getEIDsSource(adUnits) {
+  let bidderEIDSMap = new Map();
+  adUnits.forEach(adUnit => {
+    adUnit.bids.forEach(bid => {
+      let userEIDs = deepAccess(bid, 'userIdAsEids') || [];
+
+      if (userEIDs) {
+        let sourceSet = new Set();
+        for (let i = 0; i < userEIDs.length; i++) {
+          let source = userEIDs[i].source;
+          sourceSet.add(source);
+        }
+        bidderEIDSMap.set(bid.bidder, sourceSet);
+      }
+    });
+  });
+
+  return bidderEIDSMap;
+}
+
+function filterSDA(adUnits, ortb2Fragments) {
+  let bidderEIDSMap = getEIDsSource(adUnits);
+  let resetGlobal = false;
+  for (const [key, value] of Object.entries(ortb2Fragments.bidder)) {
+    let resetSDA = containsConfiguredEIDS(bidderEIDSMap, key);
+    if (resetSDA) {
+      deepSetValue(value, 'user.data', []);
+      resetGlobal = true;
+    }
+  }
+  if (resetGlobal) {
+    deepSetValue(ortb2Fragments, 'global.user.data', [])
+  }
+}
+
+function filterEIDs(adUnits, ortb2Fragments) {
+  let segementMap = getSegmentConfig(ortb2Fragments);
+  let globalEidUpdate = false;
+  adUnits.forEach(adUnit => {
+    adUnit.bids.forEach(bid => {
+      let resetEID = containsConfiguredSDA(segementMap, bid.bidder);
+      if (resetEID) {
+        globalEidUpdate = true;
+        bid.userIdAsEids = [];
+        bid.userId = {};
+        if (ortb2Fragments.bidder) {
+          let bidderFragment = ortb2Fragments.bidder[bid.bidder];
+          let userExt = deepAccess(bidderFragment, 'user.ext.eids') || [];
+          if (userExt) {
+            deepSetValue(bidderFragment, 'user.ext.eids', [])
+          }
+        }
+      }
+    });
+  });
+
+  if (globalEidUpdate) {
+    deepSetValue(ortb2Fragments, 'global.user.ext.eids', [])
+  }
+  return adUnits;
+}
+
+export function init() {
+  const confListener = config.getConfig(MODULE_NAME, dataControllerConfig => {
+    if (!dataControllerConfig || !dataControllerConfig.dataController) {
+      _logger.logInfo(`Data Controller is not configured`);
+      startAuction.getHooks({hook: filterBidData}).remove();
+      return;
+    }
+
+    if (dataControllerConfig.dataController.filterEIDwhenSDA && dataControllerConfig.dataController.filterSDAwhenEID) {
+      _logger.logInfo(`Data Controller can be configured with either filterEIDwhenSDA or filterSDAwhenEID`);
+      startAuction.getHooks({hook: filterBidData}).remove();
+      return;
+    }
+    confListener(); // unsubscribe config listener
+    _dataControllerConfig = dataControllerConfig.dataController;
```

Comment:
I think we can clean this up a little to store the dataController in a const then reference that instead or referencing the full path each time. 

i.e.
```
const dataController = dataControllerConfig && dataControllerConfig.dataController;

if (!dataController) {
    ....
}

if (dataController.filterEIDwhenSDA && dataController. filterSDAwhenEID) {
    ....
}

confListener(); // unsubscribe config listener
_dataControllerConfig = dataController;

getHook('startAuction').before(filterBidData);
```